### ISSUE TITLE
F841 detect unused variable in `with` statement

### ISSUE THREAD

**Issue by tekumara** [charliermarsh/ruff#1096]
```python
def foo():
    with open("file") as f:
        print("hello")
```

ruff doesn't detect anything but flake8 does:
```
2     F841 local variable 'f' is assigned to but never used
```

ruff 0.0.162

**Comment by charliermarsh** [comment]
This was changed intentionally in https://github.com/charliermarsh/ruff/pull/897, but I think that fix probably should've been confined to unpacking assignments.

**Comment by charliermarsh** [comment]
Will fix.
